src/client/auth/oidc.rs

```diff
@@ -23,44 +24,141 @@ use crate::{
 use super::{sasl::SaslContinue, Credential, MONGODB_OIDC_STR};
 
 /// The user-supplied callbacks for OIDC authentication.
+pub struct State {
+    callback: Callback,
+    cache: Arc<RwLock<Cache>>,
+}
+
+impl Clone for State {
+    fn clone(&self) -> Self {
+        Self {
+            callback: self.callback.clone(),
+            cache: Arc::new(RwLock::new(Cache::new())),
```

Comment:
Not sharing the cache on `clone` surprised me - where are the expected calls to `State::clone` happening?  This won't happen for `Client::clone` since the options are part of the inner shared state, and I couldn't see any obvious places in this PR (although it's hard to search for).

Reply:
I think you're probably right. I'm not sure when Credentials are generally cloned.